### PR TITLE
[CR-16240] Fix to add OFC to credit note for cancelled shipments

### PR DESCRIPTION
**Pull request by vinobiz**
https://bizongo.atlassian.net/browse/FNF-2318

Test cases:
- [ ] In case of a return shipment being cancelled, a debit note must be generated with just the product price
- [ ] In case of a forward shipment with OFC + FF charges being delivered, a CN must be generated with amount = product price and delivery amount = OFC + FF charges 
- [ ] All Shipments must be created
- [ ] Return Shipments must be created
- [ ] Shipments must be cancellable

### CODE REVIEW

File: lib/handlers/shipment_handler.rb [hunk 1]
```diff
@@ -85,14 +109,25 @@ def shipment_updated(connection, data, logger, kafka_broker)
 
     private
 
-    def create_invoice data
+    def create_invoice(data, is_return)
       common_data = InvoiceCreationHelper.common_create_invoice_data(data)
+      
+      amount = data[:shipment]['total_buyer_invoice_amount'].to_f - data[:shipment]['total_buyer_service_charge'].to_f;
```
Comment (abhijits) [inline lib/handlers/shipment_handler.rb:115]: @vinobiz if this is empty or blank? 
data[:shipment]['total_buyer_invoice_amount']

**Review by abhijits — CHANGES_REQUESTED** [review]

File: lib/handlers/shipment_handler.rb [hunk 2]
```diff
@@ -85,14 +109,27 @@ def shipment_updated(connection, data, logger, kafka_broker)
...
-    def create_invoice data
+    def create_invoice(data, is_return, logger)
       common_data = InvoiceCreationHelper.common_create_invoice_data(data)
+      
+      amount = data[:shipment]['total_buyer_invoice_amount'].to_f.round(2) - data[:shipment]['total_buyer_service_charge'].to_f.round(2);
+
+      if is_return
+        delivery_amount = 0
+      else
+        amount = amount + ((data[:shipment]['fulfilment_charges'].to_f.round(2) * APP_CONFIG["buyer_service_charges"]["rate"])/ 100 ).to_f.round(2);
```
Comment (Vishwajith-mr) [inline lib/handlers/shipment_handler.rb:120]: @abbasdawoodbiz I could not get this calculation. Can you help me understand?

Comment (abbasdawoodbiz) [reply lib/handlers/shipment_handler.rb:120]: @ishanbiz can you check if the CN created originally due to return consists of the OFC or just the tax on OFC

Comment (ishanbiz) [reply lib/handlers/shipment_handler.rb:120]: @abbasdawoodbiz none of the notes today have OFC and its tax. they only contain product value with tax.

**Pull request merged**